internal/indexmanagement/scripts.go

```diff
@@ -122,31 +123,79 @@ for i in range(0, len(indices), 25):
   print(','.join(indices[i:i+25]))
 `
 
+const indexManagement = `
+
+CONNECT_TIMEOUT=${CONNECT_TIMEOUT:-30}
+
+function getWriteIndex() {
+
+  local policy="$1"
+
+  # find out the current write index for ${POLICY_MAPPING}-write and check if there is the next generation of it
+  aliasResponse=$(curl -s $ES_SERVICE/_alias/${policy} \
+    --cacert /etc/indexmanagement/keys/admin-ca \
+    --connect-timeout ${CONNECT_TIMEOUT} \
+    -H"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    -HContent-Type:application/json \
+    --retry 5 \
+    --retry-delay 5)
+
+  if [ -z "$aliasResponse" ]; then
+    echo "Received an empty response from elasticsearch -- server may not be ready"
+    exit 1
+  fi
+
+  jsonResponse="$(echo [$aliasResponse] | sed 's/}{/},{/g')"
+
+  if ! writeIndices="$(python /tmp/scripts/getWriteIndex.py "${policy}" "$jsonResponse")" ; then
+    echo $writeIndices
+    exit 1
+  fi
+
+  writeIndex="$(ensureOneWriteIndex "$policy" "$writeIndices")"
+
+  echo $writeIndex
+}
+
+function ensureOneWriteIndex() {
+
+  local policy="$1"
+  local writeIndices="$2"
+
+  # first index received is the latest one
+  writeIndex=""
+  for index in $writeIndices; do
+    if [ -z "$writeIndex" ]; then
+      writeIndex="$index"
+    else
+      # extra write index -- mark it as not a write index
+      curl -s "$ES_SERVICE/_aliases" \
+        --connect-timeout ${CONNECT_TIMEOUT} \
+        --cacert /etc/indexmanagement/keys/admin-ca \
+        -HContent-Type:application/json \
+        -XPOST \
+        -H"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+        -o /dev/null \
+        -d '{"actions":[{"add":{"index": "'$index'", "alias": "'${POLICY_MAPPING}-write'", "is_write_index": false}}]}' \
+        --retry 5 \
+        --retry-delay 5
+    fi
+  done
+
+  echo $writeIndex
+}
+`
+
 const rolloverScript = `
 set -euo pipefail
-CONNECT_TIMEOUT=${CONNECT_TIMEOUT:-30}
+source /tmp/scripts/indexManagement
+
 decoded=$(echo $PAYLOAD | base64 -d)
 
 echo "Index management rollover process starting"
 
 # get current write index
-# find out the current write index for ${POLICY_MAPPING}-write and check if there is the next generation of it
-writeIndices=$(curl -s $ES_SERVICE/_alias/${POLICY_MAPPING}-write \
-  --cacert /etc/indexmanagement/keys/admin-ca \
-  --connect-timeout ${CONNECT_TIMEOUT} \
-  -H"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
-  -HContent-Type:application/json \
-  --retry 5 \
-  --retry-delay 5)
-
-if [ -z "$writeIndices" ]; then
-  echo "Received an empty response from elasticsearch -- server may not be ready"
-  exit 1
-fi
-
-jsonResponse="$(echo [$writeIndices] | sed 's/}{/},{/g')"
-
-if ! writeIndex="$(python /tmp/scripts/getWriteIndex.py "${POLICY_MAPPING}-write" "$jsonResponse")" ; then
+if ! writeIndex="$(getWriteIndex "${POLICY_MAPPING}-write")" ; then
```

Comment:
Can we spell out the `echo $writeIndex` as a proper exit message to understand what failed where?